app/scripts/src/storage.es.js

```diff
@@ -48,36 +45,27 @@ module.factory('languageStorage', [
             }
         };
 
-        var determineStartingLanguage = function () {
-            // First we check to see if a cookie language exists, then we see if its valid.
-            // If valid we make it the selected language.
-            var cookieLanguage = $cookies.get('selectedLanguage');
-            var browserLanguage = $window.navigator.language || $window.navigator.userLanguage;
-
-            var choice;
-            if (angular.isDefined(cookieLanguage)) {
-                choice = languageStorage.getLanguageChoice(cookieLanguage);
-                if (angular.isDefined(choice)) {
-                    // We found a valid language in the cookie, so use that
-                    return choice.id;
-                }
-            } else if (angular.isDefined(browserLanguage)) {
-                // if the cookie doesn't exist/invalid then check language from the browser to
-                // set as default.
-                choice = languageStorage.getLanguageChoice(browserLanguage);
-                if (angular.isDefined(choice)) {
-                    // We found a valid language in the cookie, so use that
-                    return choice.id;
-                }
+        const checkLanguage = function (language) {
+            if (angular.isDefined(language)) {
+                choice = publicMethods.getLanguageChoice(language);
```

Comment:
I think the conditional here can be explained more in context of `getLanguageChoice: function (languageId = selectedLanguage) {}`, in that it's required so that this check ignores the default language.

Maybe make another `getLanguageChoice` method that does not use the default language? `getLanguageChoiceWithoutDefault`? Then you can have:
```js
const checkLanguage = function (language) {
    const choice = publicMethods.getLanguageChoiceWithoutDefault(language);
    return choice && choice.id;
};
```

Also, given its use, I think `checkLanguage` needs to be renamed, since it's returning a language ID if it can.